hack/jenkins/job-configs/kubernetes-e2e.yaml

```diff
@@ -358,13 +358,10 @@
             test-owner: 'beeps'
             provider-env: '{gke-provider-env}'
             job-env: |
-                # XXX Not a unique project
                 export E2E_NAME="e2e-gke-ingress"
```

Comment:
Having stuff like this in the config more tightly couples the config to the tests, making it harder to refactor, etc.

`E2E_NAME` doesn't provide any information—it just feeds into what we call the cluster & network to differentiate it from other clusters running in the same project, which this is no longer doing, and the vast majority of people shouldn't have to think about GCP resource leaks at all—if that kind of thing fails, this config isn't the first place you'll look, and if you're looking at this config, you probably don't care that we check resource leakages, right?